### PR TITLE
Changed Simulation and Reports to take more advantage of libsonata

### PR DESCRIPTION
**Pull request by joni-herttuainen**

### CODE REVIEW

File: bluepysnap/frame_report.py
Comment (mgeplf) [reply bluepysnap/frame_report.py:44]: seems like a libsonata bug? can you open an issue in that repo?

Comment (joni-herttuainen) [reply bluepysnap/frame_report.py:44]: Sure thing: https://github.com/BlueBrain/libsonata/issues/221

File: bluepysnap/simulation.py
Comment (GianlucaFicarelli) [inline bluepysnap/simulation.py:95]: I don't see `tstart` in https://github.com/BlueBrain/libsonata/blob/e0cdaf707feae298d432a11742d2620721a1ae10/python/bindings.cpp#L530
or https://github.com/BlueBrain/libsonata/blob/e0cdaf707feae298d432a11742d2620721a1ae10/include/bbp/sonata/config.h#L192
and it's not even defined at https://sonata-extension.readthedocs.io/en/latest/sonata_simulation.html#run
🤔

File: bluepysnap/simulation.py
Comment (GianlucaFicarelli) [inline bluepysnap/simulation.py:116]: With this change, `conditions` would return `None` instead of `{}` when the attribute is missing.
Is it intentional? In theory it could break code using it.

File: tests/test_simulation.py
Comment (GianlucaFicarelli) [inline tests/test_simulation.py:87]: The test is now different and should be renamed from `test_no_network_config` to something like `test_network_file_not_found`.
However, maybe the old test `test_no_network_config` could be kept to verify also that case.

File: tests/test_simulation.py
Comment (GianlucaFicarelli) [inline tests/test_simulation.py:95]: Why the file `circuit_config.json` needs to be removed to make the test succeed?
Isn't it related to the configuration only?

**Comment by GianlucaFicarelli** [comment]
The new changes can be mentioned more explicitly in the changelog.

File: bluepysnap/simulation.py
Comment (joni-herttuainen) [reply bluepysnap/simulation.py:78]: [`libsonata`](https://github.com/BlueBrain/libsonata/blob/master/src/config.cpp#L933) defaults `network` to `{base_path}/circuit_config.json` if not found.


I don't know if it'd be better to remove the default from `libsonata` and just do
```python
return getattr(self.to_libsonata, 'network', None)
```
or revert to the original checking of the actual value in config or what. 

Any thoughts, @mgeplf / @jdcourcol ?

File: bluepysnap/simulation.py
Comment (joni-herttuainen) [reply bluepysnap/simulation.py:95]: You're right. Me neither, didn't even check this. Just assumed this was correct in the first place. I wonder what's the correct/intended behavior here? :thinking:  

Should we even have `time_start` in the `Simulation` class to begin with? :man_shrugging:

File: bluepysnap/simulation.py
Comment (joni-herttuainen) [reply bluepysnap/simulation.py:116]: Yeah, `libsonata` returns an object (`libsonata._libsonata.Conditions`) if it was defined and I felt it would be counter-intuitive to return empty dictionary by default.

About breaking changes, this would break the code using it, anyway. We're releasing a new major version, so I am not too worried about small breaking changes. 

I agree, I should mention these in the changelog. I'll update it.

File: tests/test_simulation.py
Comment (joni-herttuainen) [reply tests/test_simulation.py:95]: That is, again, due to [`libsonata`](https://github.com/BlueBrain/libsonata/blob/master/src/config.cpp#L944): If `node_sets_file` is not defined in the simulation config, `libsonata` fetches it from the circuit config. 

Technically, the correct approach would be to remove it also from the circuit config but this was just simpler. I'll clarify it with a comment.

File: tests/test_simulation.py
Comment (joni-herttuainen) [reply tests/test_simulation.py:87]: I'll rename the test if need be. We should discuss and decide on the libsonata behavior first, though.

File: bluepysnap/simulation.py
Comment (GianlucaFicarelli) [reply bluepysnap/simulation.py:78]: ic, it seems ok to me to keep your changes and relying on libsonata using "circuit_config.json" as default, since it's mentioned in the sonata specs.

File: tests/test_simulation.py
Comment (GianlucaFicarelli) [reply tests/test_simulation.py:87]: Yeah, no need to add `test_no_network_config` as I was thinking in the previous comment. I would just rename the name of the test.

File: bluepysnap/simulation.py
Comment (joni-herttuainen) [reply bluepysnap/simulation.py:78]: So it seems. I'll keep this behavior, then

File: tests/test_simulation.py
Comment (joni-herttuainen) [reply tests/test_simulation.py:87]: > renamed

File: bluepysnap/simulation.py
Comment (mgeplf) [reply bluepysnap/simulation.py:95]: Open a ticket in libsonata if there is an inconsistency, pls; don't forget to look in https://github.com/AllenInstitute/sonata/blob/master/docs/SONATA_DEVELOPER_GUIDE.md as well

Comment (joni-herttuainen) [reply bluepysnap/simulation.py:95]: Okay. According to jdc, simulation should always start at `0` so this is not in the spec.
I'll have this return 0, then.

**Review by GianlucaFicarelli — APPROVED** [review]
LGTM, unless it can wait for the release of `libsonata>=0.1.14` to adapt the retrieval of the filename

**Comment by joni-herttuainen** [comment]
Yeah, I'll just add that to my list and fix it when the next version of libsonata is released. There probably wouldn't be any merge conflicts even if I waited but just in case

**Pull request merged**
